File: Java/DesignPatterns/src/main/java/designPatterns/behavioral/mediator/ConcreteMediator.java
package designPatterns.behavioral.mediator;

import java.util.ArrayList;
import java.util.List;

public class ConcreteMediator implements Mediator {
  private List<Model> models;

  public ConcreteMediator() {
    models = new ArrayList<>();
  }

  @Override
  public void sendMessage(String msg, Model model) {
   for (Model m : models) {
     if (m != model) m.receive(msg);
   }
  }

  @Override
  public boolean addModel(Model model) {
    return models.add(model);
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/behavioral/mediator/ConcreteMediator.kt
package behavioral.mediator

class ConcreteMediator : Mediator {
  private var models: MutableList<Model>

  init {
    models = ArrayList()
  }

  override fun sendMessage(msg: String, model: Model) {
    for (m in models) {
      if (m != model) {
        m.receive(msg)
      }
    }
  }

  override fun addModel(model: Model): Boolean = models.add(model)
}